engine/paymail/paymail_service_client.go

```diff
@@ -0,0 +1,194 @@
+package paymail
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/bitcoin-sv/go-paymail"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/mrz1836/go-cachestore"
+	"github.com/rs/zerolog"
+)
+
+const cacheKeyCapabilities = "paymail-capabilities-"
+const cacheTTLCapabilities = 60 * time.Minute
+
+type service struct {
+	cache         cachestore.ClientInterface
+	paymailClient paymail.ClientInterface
+	log           zerolog.Logger
+}
+
+// NewServiceClient creates a new paymail service client
+func NewServiceClient(cache cachestore.ClientInterface, paymailClient paymail.ClientInterface, log zerolog.Logger) ServiceClient {
+	if paymailClient == nil {
+		panic(spverrors.Newf("paymail client is required to create a new paymail service"))
+	}
+
+	if cache == nil {
+		log.Info().Msg("Doesn't receive cachestore, won't use cache for capabilities")
+	}
+
+	return &service{
+		cache:         cache,
+		paymailClient: paymailClient,
+		log:           log,
+	}
+}
+
+// GetSanitizedPaymail validates and returns the sanitized version of paymail address (alias@domain.tld)
+func (s *service) GetSanitizedPaymail(addr string) (*paymail.SanitisedPaymail, error) {
+	if err := paymail.ValidatePaymail(addr); err != nil {
+		return nil, err //nolint:wrapcheck // we have handler for paymail errors
+	}
+
+	sanitized := &paymail.SanitisedPaymail{}
+	sanitized.Alias, sanitized.Domain, sanitized.Address = paymail.SanitizePaymail(addr)
+
+	return sanitized, nil
+}
+
+// GetCapabilities is a utility function to retrieve capabilities for a Paymail provider
+func (s *service) GetCapabilities(ctx context.Context, domain string) (*paymail.CapabilitiesPayload, error) {
+	// todo: allow user to configure the time that they want to cache the capabilities (if they want to cache or not)
+
+	cacheKey := cacheKeyCapabilities + domain
+
+	capabilities, err := s.loadCapabilitiesFromCache(ctx, cacheKey)
+	if err != nil {
+		return nil, err
+	}
+	if capabilities != nil {
+		return capabilities, err
+	}
+
+	response, err := s.loadCapabilities(domain)
+	if err != nil {
+		return nil, err
+	}
+
+	s.putCapabilitiesInCache(ctx, cacheKey, response.CapabilitiesPayload)
+
+	return &response.CapabilitiesPayload, nil
+}
+
+func (s *service) loadCapabilitiesFromCache(ctx context.Context, key string) (*paymail.CapabilitiesPayload, error) {
+	if s.cache == nil {
+		return nil, nil
+	}
+
+	capabilities := new(paymail.CapabilitiesPayload)
+	err := s.cache.GetModel(ctx, key, capabilities)
+	if errors.Is(err, cachestore.ErrKeyNotFound) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, spverrors.Wrapf(err, "failed to get capabilities from cachestore")
+	}
```

Comment:
```suggestion
	
	if err != nil {
	        if errors.Is(err, cachestore.ErrKeyNotFound) {
		        return nil, nil
	        }
		return nil, spverrors.Wrapf(err, "failed to get capabilities from cachestore")
	}
```

Or you did it on purpose - in that case - leave it as is 😄

Reply:
I did it in purpose, because `errors.Is(nil, cachestore.ErrKeyNotFound)` will be false, and it's easier to reason and read without this wrapping